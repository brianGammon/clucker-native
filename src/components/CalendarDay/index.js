import CalendarDayComponent from './CalendarDay';

export default CalendarDayComponent;
