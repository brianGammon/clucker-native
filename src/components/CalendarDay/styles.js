import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  details: {
    flex: 1,
  },
  badgeContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  badgeText: {
    fontSize: 12,
  },
  notesContainer: {
    flexDirection: 'row',
    marginTop: 10,
  },
  label: {
    fontSize: 17,
    fontWeight: 'bold',
  },
  eggNote: {
    fontStyle: 'italic',
    color: 'grey',
    marginLeft: 10,
    padding: 5,
  },
  moreButton: {
    alignSelf: 'center',
  },
  moreIcon: {
    fontSize: 30,
  },
});
