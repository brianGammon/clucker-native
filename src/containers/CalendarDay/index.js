/* @flow */
import React from 'react';
import { Text, Button, View } from 'react-native';

export interface Props {
  navigation: any;
}

export default class CalendarDay extends React.Component<Props> {
  render() {
    const { navigation } = this.props;
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Text style={{ fontSize: 30 }}>Todays Eggs</Text>
        <Button onPress={() => navigation.goBack()} title="Back to Month" />
      </View>
    );
  }
}
