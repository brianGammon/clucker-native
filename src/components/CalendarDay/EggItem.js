/* @flow */
import React from 'react';
import {
  View, Text, ListItem, Body, Button, Icon, Badge,
} from 'native-base';
import { type Egg } from '../../types';
import CommonLabel from '../CommonLabel';
import styles from './styles';

type Props = {
  item: string,
  chickenName: string,
  egg: Egg,
  handleMoreOptions: (eggId: string, bulkMode: boolean) => void,
};

const EggItem = ({
  item, egg, chickenName, handleMoreOptions,
}: Props) => (
  <ListItem>
    <Body style={styles.container}>
      <View style={styles.flex}>
        <View style={{ flexDirection: 'row', alignItems: 'center' }}>
          {!egg.bulkMode && (
            <View style={styles.flex}>
              <Text style={styles.label}>{chickenName}</Text>
              <View style={styles.row}>
                <CommonLabel text="Weight: " />
                <Text>{`${egg.weight || '-- '}g`}</Text>
              </View>
            </View>
          )}
          {egg.bulkMode && (
            <View style={styles.flex}>
              <Text style={styles.label}>Bulk Entry</Text>
              <View style={styles.row}>
                <CommonLabel text="Quantity: " />
                <Text>{egg.quantity}</Text>
              </View>
            </View>
          )}
          {egg.damaged && (
            <View style={styles.badgeContainer}>
              <Badge danger>
                <Text style={styles.badgeText}>Damaged</Text>
              </Badge>
            </View>
          )}
        </View>
        {!!egg.notes
          && egg.notes !== '' && (
            <View style={styles.notesContainer}>
              <Icon name="attach" />
              <Text style={styles.eggNote}>{egg.notes}</Text>
            </View>
        )}
      </View>
      <View>
        <Button
          style={styles.moreButton}
          transparent
          dark
          onPress={() => handleMoreOptions(item, egg.bulkMode)}
        >
          <Icon active style={styles.moreIcon} name="more" />
        </Button>
      </View>
    </Body>
  </ListItem>
);

export default EggItem;
