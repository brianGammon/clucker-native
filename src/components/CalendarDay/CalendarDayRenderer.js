/* @flow */
import * as React from 'react';
import { Container, Content } from 'native-base';
import { FlatList } from 'react-native';
import { type Egg, type Chicken } from '../../types';
import Header from '../Header';
import DateSwitcher from '../DateSwitcher';
import Line from '../Line';
import EggItem from './EggItem';
import { UNNAMED_HEN_LABEL } from '../../constants';

type Props = {
  navigation: any,
  dates: {
    now: string,
    date: string,
    previousDate: string,
    nextDate?: string,
  },
  eggs: {
    [eggId: string]: Egg,
  },
  count: number,
  chickens: {
    [chickenId: string]: Chicken,
  },
  handleMoreOptions: (eggId: string, bulkMode: boolean) => void,
};

const CalendarDayRenderer = ({
  navigation,
  eggs,
  chickens,
  dates,
  handleMoreOptions,
  count,
}: Props) => (
  <Container>
    <Header title="Daily Log" eggButton goBackButton="Month" />
    <Content padder>
      <DateSwitcher
        mode="day"
        navigation={navigation}
        dates={dates}
        eggCount={count}
      />
      <Line />
      <FlatList
        data={Object.keys(eggs || {})}
        keyExtractor={(item, index) => index.toString()}
        renderItem={({ item }) => (
          <EggItem
            item={item}
            chickenName={
              (chickens[eggs[item].chickenId]
                && chickens[eggs[item].chickenId].name)
              || UNNAMED_HEN_LABEL
            }
            egg={eggs[item]}
            handleMoreOptions={handleMoreOptions}
          />
        )}
      />
    </Content>
  </Container>
);

export default CalendarDayRenderer;
