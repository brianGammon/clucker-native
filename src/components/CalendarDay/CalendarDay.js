/* @flow */
import * as React from 'react';
import { connect } from 'react-redux';
import { Alert } from 'react-native';
import CalendarDayRenderer from './CalendarDayRenderer';
import { nowAsMoment, dateSwitcher } from '../../utils/dateHelper';
import eggsByRangeSelector from '../../selectors/eggsByRangeSelector';
import { metaTypes } from '../../redux/constants';
import { firebaseRemoveRequested } from '../../redux/actions';
import { type Egg } from '../../types';

type Props = {
  navigation: any,
  flockId: string,
  dates: {
    date: string,
    previousDate: string,
    nextDate?: string,
  },
  eggs: {
    [eggId: string]: Egg,
  },
  deleteEgg: (flockId: string, eggId: string) => void,
};

class CalendarDay extends React.Component<Props> {
  onDeleteEgg = (eggId) => {
    const { flockId, deleteEgg } = this.props;
    Alert.alert('Are you sure you want to delete this egg?', null, [
      {
        text: 'Cancel',
      },
      {
        text: 'Delete',
        onPress: () => deleteEgg(flockId, eggId),
        style: 'destructive',
      },
    ]);
  };

  render() {
    const { navigation, dates, eggs } = this.props;
    return (
      <CalendarDayRenderer
        navigation={navigation}
        dates={dates}
        eggs={eggs}
        onDeleteEgg={this.onDeleteEgg}
      />
    );
  }
}

const mapStateToProps = ({ eggs, userSettings }, { navigation }) => {
  const date = navigation.getParam('date', nowAsMoment().format('YYYY-MM-DD'));
  const { previousDate, nextDate } = dateSwitcher(date, 'days', 'YYYY-MM-DD');
  return {
    dates: {
      date,
      previousDate,
      nextDate,
    },
    flockId: userSettings.data.currentFlockId,
    eggs: eggsByRangeSelector(eggs.data, date),
  };
};

const mapDispatchtoProps = dispatch => ({
  deleteEgg: (flockId, eggId) => dispatch(firebaseRemoveRequested({ flockId, eggId }, metaTypes.eggs)),
});

export default connect(
  mapStateToProps,
  mapDispatchtoProps,
)(CalendarDay);
