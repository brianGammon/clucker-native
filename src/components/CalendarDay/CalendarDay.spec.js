import React from 'react';
import CalendarDay from '.';

describe('CalendarDay component:', () => {
  test('Should render correctly', () => {
    const wrapper = shallow(<CalendarDay />);
    expect(wrapper).toMatchSnapshot();
  });
});
