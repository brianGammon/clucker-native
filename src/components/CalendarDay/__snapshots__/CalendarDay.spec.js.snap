// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDay component: Should render correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "fontSize": 30,
      }
    }
  >
    Todays Eggs
  </Component>
  <Button
    onPress={[Function]}
    title="Back to Month"
  />
</Component>
`;
